contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -868,10 +911,16 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
      *         call GetLastError.
      */
     public static final String QueryFullProcessImageName(HANDLE hProcess, int dwFlags) {
-        char[] path = new char[WinDef.MAX_PATH];
-        IntByReference lpdwSize = new IntByReference(path.length);
-        if (Kernel32.INSTANCE.QueryFullProcessImageName(hProcess, 0, path, lpdwSize))
-            return new String(path).substring(0, lpdwSize.getValue());
+        int size = 0;
+        IntByReference lpdwSize = new IntByReference();
+        do {
+            size = size + 1024;
```

Comment:
Since `WinDef.MAX_PATH` should work in most cases, let's use that as the initial size.  You can initialize on line 914 and increment (I'm ok with += 1024) after the query conditional.